plugin/storage/integration/grpc_server.go

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package integration
+
+import (
+	"net"
+	"sync"
+
+	googleGRPC "google.golang.org/grpc"
+
+	grpcMemory "github.com/jaegertracing/jaeger/plugin/storage/grpc/memory"
+	"github.com/jaegertracing/jaeger/plugin/storage/grpc/shared"
+	"github.com/jaegertracing/jaeger/plugin/storage/memory"
+	"github.com/jaegertracing/jaeger/ports"
+)
+
+type GRPCServer struct {
+	errChan chan error
```

Comment:
Is this actually needed? If it is, I would prefer to use a callback function, not a channel, because channels like this tend to cause deadlocks when nothing is reading them, while the callback achieves the same result of communicating the status without the risk of blocking (and the implementation of the callback can still use channel if it wants to, but that channel will be more local and less prone to deadlocks)

Reply:
I see, I have updated this to pass the error through the struct field